File: src/RequestType.java
import java.util.Optional;
import static java.util.Optional.*;

public enum RequestType {
    GET,
    POST;

    static Optional<RequestType> forString(String string) {
        if (string.equals("GET"))
            return of(RequestType.GET);
        else if (string.equals("POST"))
            return of(RequestType.POST);
        else
            return empty();
    }
}

File: src/RequestType.kt

enum class RequestType {
    GET,
    POST;

    companion object {

        internal fun forString(string: String): RequestType? {
            return when (string) {
                "GET" -> RequestType.GET
                "POST" -> RequestType.POST
                else -> null
            }
        }
    }
}
